src/gc.c

```diff
@@ -3181,6 +3181,8 @@ JL_DLLEXPORT int64_t jl_gc_sync_total_bytes(int64_t offset) JL_NOTSAFEPOINT
 
 JL_DLLEXPORT int64_t jl_gc_pool_live_bytes(void)
 {
+    gc_n_threads = jl_atomic_load_acquire(&jl_n_threads);
+    gc_all_tls_states = jl_atomic_load_relaxed(&jl_all_tls_states);
```

Comment:
Do these need to be globals?

Reply:
Good catch. It is a data race to update these globals here

Reply:
> Do these need to be globals?

Probably not. Will make them local.